client/lib/browser-storage/index.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * Internal dependencies
+ */
+import {
+	getStoredItem as bypassGet,
+	setStoredItem as bypassSet,
+	clearStorage as bypassClear,
+} from './bypass';
+
+let shouldBypass = false;
+
+const DB_NAME = 'calypso';
+const DB_VERSION = 2; // Match versioning of the previous localforage-based implementation.
+const STORE_NAME = 'calypso_store';
+
+const supportsIDB = typeof window !== 'undefined' && window.indexedDB;
+
+let cachedDbPromise: Promise< IDBDatabase > | null = null;
+
+function prepareIDB() {
+	const request = indexedDB.open( DB_NAME, DB_VERSION );
```

Comment:
Can the `indexedDB.open` call ever throw an exception? Then it's better to put it into the Promise initializer. Exception then results in a rejected promise rather than throwing from `prepareIDB`.

Reply:
According to MDN, any errors that occur during opening result in an `error` event rather than an exception.